packages/react/src/components/Menu/Menu.tsx

```diff
@@ -124,7 +124,7 @@ class Menu extends AutoControlledComponent<WithAsProp<MenuProps>, MenuState> {
   static Item = MenuItem
   static Divider = MenuDivider
 
-  handleItemOverrides = predefinedProps => ({
+  handleItemOverrides = variables => predefinedProps => ({
```

Comment:
may suggest to remove `functionFactory -> function` relationship, as well as potential struggling points for the code reader while trying to understand what each function does given its name only.

We may remove currying altogether, and leave just two functions with `override` semantics - one for `menuItem`, another for `divider`. Will take `menuItem` as an example:

```jsx
// was: handleItemOverrides = variables => predefinedProps => ({
overrideItemProps = (predefinedProps, variables) => ({ ... })
```

And then, at the place where is used:
```jsx
// renderItems() method, we have 'variables' in scope here
MenuItem.create(item, {
    defaultProps: {...}
    overrideProps: props => this.overrideItemProps(props, variables),
})
```

Reply:
Can't agree with this, `Dropdown` component does the same thing and it looks obscure when `props` are passed in different order. Benefit of the factory: it returns the function with the same signature as `overrideProps` are defined.

Reply:
Lets focus on this specific example - does suggestion make things obscure? I think that not.

To my mind, it is much more obscure when we have several abstraction layers for, essentially, the same thing - and with this approach we have only one function, which is explicit on the input data necessary and _is not dependent on the `overrideProps` signature_ - which, in fact, it shouldn't be dependent on. 

I would see benefits of currying if this function will be reused in multiple places, but it is not - thus, with current example cannot see any reason for this complication.

Reply:
Discussed offline. Let's leave it as it is (for now) to be consistent with other components (Dropdown)